scripts/dts/extract/compatible.py

```diff
@@ -54,6 +54,17 @@ def extract(self, node_address, prop, def_label):
                 }
                 insert_defs(node_address, load_defs, {})
 
+        # Generate defines of the form:
+        # #define DT_<COMPAT>_<INSTANCE ID> 1
```

Comment:
I'm not actually sure, the bus label #define is used today as a compile switch to know how to compile driver for sensors compatible with multiple bus supported (say i2c or spi). As soon as one instance is there, "define should be generated so matching driver is compiled, whatever the instance in use (cf #12105).
This being said, this deserves a comment.

Reply:
It is possible to support multiple instances of the same sensor on different bus types, even if no in-tree code chooses to do so.  I wouldn't want the infrastructure to artificially reduce a system designer's options.

Reply:
Sure, but this specifc bus tye #define was made to help compilation of the adequate variant of the driver. You need to compile sensor_spi.c whatever number of spi sensors are there. This is not instance dependant.

Reply:
Sure, but this specifc bus tye #define was made to help compilation of the adequate variant of the driver. You need to compile sensor_spi.c whatever number of spi sensors are there. This is not instance dependant.

Reply:
That's a good point, but the driver still needs to know whether DT_COMPAT_INSTANCE_ should be defined to use the I2C or the SPI API.  So if the unqualified `DT_COMPAT_BUS_type` is used to enable/disable the bus-specific driver, there still needs to be an instance-qualified version to select the bus for that instance.

Reply:
> That's a good point, but the driver still needs to know whether DT_COMPAT_INSTANCE_ should be defined to use the I2C or the SPI API. 

Right, but we don't have this code there yet, so I agree information will be needed, but we don't know how.

> So if the unqualified `DT_COMPAT_BUS_type` is used to enable/disable the bus-specific driver, there still needs to be an instance-qualified version to select the bus for that instance.

Then we might need to generate another more explicit define:
DT_<COMPAT>_<INSTANCE>_BUS_<BUS_TYPE>

Reply:
I don't understand the second part of your response; for the first if we generate both the instance-qualified and non-instance-qualified bus defines then it's pretty likely the information we need will be available and we can make something work.  If we don't have the instance-qualified one we definitely can't make it work.

Reply:
Yeah, sorry, I was betrailled by markdown and I was maybe not clear enough.
So I meant rather than:
DT_COMPAT_BUS_\<TYPE\>_INSTANCE
generate:
DT_COMPAT_INSTANCE_BUS_\<TYPE\>

Reply:
> Is that OK?

Yep!

Reply:
Done